### PR TITLE
Doco for ASQ custom envelope unwrapper

### PR DESCRIPTION
**Pull request by andreasohlund**
First part of https://github.com/Particular/docs.particular.net/issues/2389

@Particular/azure-maintainers please review

### CODE REVIEW

File: nservicebus/azure-storage-queues/configuration.md [hunk 1]
```diff
@@ -102,3 +102,13 @@ partial: aliasesAndHashing
 ## Serialization
 
 Azure Storage Queues Transport changes the default serializer to JSON. The serializer can be changed using the [serialization API](/nservicebus/serialization).
+
+## Custom envelope unwrapper
+
+Azure Storage Queues lacks native header support. NServiceBus solves this by wrapping headers and message body in a custom envelope structure. This envelope is serialized using the configured [serializer](/nservicebus/serialization) for the endpoint before being sent.
+
+Creating this envelope can cause uneeded complexity should headers not be needed for example in native integration scenarios. For this reason Version 7.1 and onwards support configuring a custom envelop unwrapper.
```
Comment (SeanFeldman) [inline nservicebus/azure-storage-queues/configuration.md]: > Version 7.1 and onwards

Version 7.1 and **above**

File: nservicebus/azure-storage-queues/configuration.md [hunk 2]
```diff
@@ -102,3 +102,13 @@ partial: aliasesAndHashing
...
 
 Azure Storage Queues Transport changes the default serializer to JSON. The serializer can be changed using the [serialization API](/nservicebus/serialization).
+
+## Custom envelope unwrapper
+
+Azure Storage Queues lacks native header support. NServiceBus solves this by wrapping headers and message body in a custom envelope structure. This envelope is serialized using the configured [serializer](/nservicebus/serialization) for the endpoint before being sent.
+
+Creating this envelope can cause uneeded complexity should headers not be needed for example in native integration scenarios. For this reason Version 7.1 and onwards support configuring a custom envelop unwrapper. 
+
+The snippet below shows an example of a custom unwrapper that enables both NServiceBus formatted and plain JSON messages to be consumed.
```
Comment (SeanFeldman) [inline nservicebus/azure-storage-queues/configuration.md]: >  example of a custom unwrapper 

Perhaps "custom unwrapping logic"?

File: Snippets/ASQ/ASQ_7/CustomEnvelopeUnwrapper.cs [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        transport.UnwrapMessagesWith(cloudQueueMessage =>
+        {
+            using (var stream = new MemoryStream(cloudQueueMessage.AsBytes))
+            using (var streamReader = new StreamReader(stream))
+            using (var textReader = new JsonTextReader(streamReader))
+            {
+                //try deserialize to a NServiceBus envelope first
+                var wrapper = jsonSerializer.Deserialize<MessageWrapper>(textReader);
+
+                if (!string.IsNullOrEmpty(wrapper.Id))
```
Comment (SimonCropp) [inline Snippets/ASQ/ASQ_7/CustomEnvelopeUnwrapper.cs]: is empty really a valid value?

File: Snippets/ASQ/ASQ_7/project.json [hunk 4]
```diff
@@ -1,7 +1,8 @@
 ﻿{
   "dependencies": {
-    "NServiceBus.Azure.Transports.WindowsAzureStorageQueues": "7.*",
-    "NServiceBus.Host": "7.*"
+    "NServiceBus.Azure.Transports.WindowsAzureStorageQueues": "7.1.0",
```
Comment (SimonCropp) [inline Snippets/ASQ/ASQ_7/project.json]: why not wildcards?

**Comment by andreasohlund** [comment]
Just did a nuget update, will fix

File: Snippets/ASQ/ASQ_7/CustomEnvelopeUnwrapper.cs (hunk 3 above)
Comment (andreasohlund) [inline Snippets/ASQ/ASQ_7/CustomEnvelopeUnwrapper.cs]: Empty isn't valid so we don't return? (are you saying this should be `!= null` ?)

File: Snippets/ASQ/ASQ_7/project.json (hunk 4 above)
Comment (andreasohlund) [inline Snippets/ASQ/ASQ_7/project.json]: Just did a nuget update, will fix

File: Snippets/ASQ/ASQ_7/CustomEnvelopeUnwrapper.cs (hunk 3 above)
Comment (SimonCropp) [inline Snippets/ASQ/ASQ_7/CustomEnvelopeUnwrapper.cs]: That is what I was alluding to

**Comment by andreasohlund** [comment]
Review comments should now be addressed

**Pull request merged**